kubernetes/kubernetes_meter.py

```diff
@@ -0,0 +1,176 @@
+        # TODO the following fields are not currently tracked:
+        #      memory, machine name, grid
+        r = Gratia.UsageRecord(RESOURCE_TYPE)
+        r.StartTime(   self.getint('StartTime'), SECONDS)
+        r.MachineName (self.get('MachineName'))
+        r.LocalJobId(  self.get('LocalJobId'))
+        r.EndTime(     self.getint('EndTime'), SECONDS)
+        r.WallDuration(self.getint('WallDuration'), SECONDS)
+        r.CpuDuration( self.getint('CpuDuration'), USER, SECONDS)
+        r.Memory(      self.getint('MemoryVirtual'), 'KB', description='RSS')
+        r.Processors(  self.getint('Processors'), metric="max")
+        r.SiteName(    self.get('Site'))
+        r.ProbeName(   self.site_probe())
+        r.Grid(        self.get('InfrastructureType')) # Best guess
+        r.LocalUserId( LOCAL_USER)
+        r.VOName(      self.get('VO'))
+        r.ReportableVOName(self.get('VO'))
+        return r
+
+    @classmethod
+    def is_individual_record(cls, data:bytes):
+        return data.decode().startswith('APEL-individual-job-message:')
+
+
+def send_gratia_records(records: list[ApelRecordConverter]):
+    # TODO the assumption of uniform site/probe might not be true
+    site = records[0].get('Site')
+    probe = records[0].site_probe()
+
+    # GratiaCore.Initialize(gratia_config)
+
+    config.Config.setSiteName(site)
+    config.Config.setMeterName(probe)
+
+    GratiaCore.Handshake()
+
+    try:
+        GratiaCore.SearchOutstandingRecord()
+    except Exception as e:
+        print(f"Failed to search outstanding records: {e}")
+        raise
+
+    GratiaCore.Reprocess()
+
+    for record in records:
+        resp = GratiaCore.Send(record.to_gratia_record())
+        print(resp)
+
+    GratiaCore.ProcessCurrentBundle()
+            
+
+
+
+def setup_gratia(config: GratiaK8sConfig):
+
+    print(config.gratia_config_path)
+    if not config.gratia_config_path or not os.path.exists(config.gratia_config_path):
+        raise Exception("No valid gratia config path given")
+    GratiaCore.Config = GratiaCore.ProbeConfiguration(config.gratia_config_path)
+
+    GratiaWrapper.CheckPreconditions()
+    GratiaWrapper.ExclusiveLock()
+
+    # Register gratia
+    GratiaCore.RegisterReporter("kubernetes_meter")
+    GratiaCore.RegisterService("Kubernetes", config.gratia_probe_version)
+    GratiaCore.setProbeBatchManager("kubernetes")
+
+    GratiaCore.Initialize(config.gratia_config_path)
+
+def batch_dirq(queue, batch_size):
+    """ batch the records in a dirq into groups of a fixed size """
+    # TODO this is not the most elegant approach
+    records = []
+    for name in queue:
+        records.append(name)
+        if len(records) >= batch_size:
+            yield records
+            records = []
+    # Yield the last entries as well
+    yield records
+
+def main(envFile: str):
+    print(f'Starting Gratia post-processor: {__file__} with envFile {envFile} at {datetime.now(tz=timezone.utc).isoformat()}')
```

Comment:
Can we also dump the contents of the env file? It could be helpful for debugging.